src/std_extensions/arithmetic/int_ops.rs

```diff
@@ -40,100 +41,145 @@ fn inarrow_sig(arg_values: &[TypeArg]) -> Result<FunctionType, SignatureError> {
     ))
 }
 
-fn itob_sig(_arg_values: &[TypeArg]) -> Result<FunctionType, SignatureError> {
-    Ok(FunctionType::new(
-        vec![int_type(type_arg(0))],
-        type_row![BOOL_T],
-    ))
+fn int_polytype(
+    n_vars: usize,
+    input: impl Into<TypeRow>,
+    output: impl Into<TypeRow>,
+    temp_reg: &ExtensionRegistry,
+) -> Result<PolyFuncType, SignatureError> {
+    PolyFuncType::new_validated(
+        vec![LOG_WIDTH_TYPE_PARAM; n_vars],
+        FunctionType::new(input, output),
+        temp_reg,
+    )
 }
 
-fn btoi_sig(_arg_values: &[TypeArg]) -> Result<FunctionType, SignatureError> {
-    Ok(FunctionType::new(
-        type_row![BOOL_T],
-        vec![int_type(type_arg(0))],
-    ))
+fn itob_sig(
+    int_type_var: Type,
+    temp_reg: &ExtensionRegistry,
+) -> Result<PolyFuncType, SignatureError> {
+    int_polytype(1, vec![int_type_var], type_row![BOOL_T], temp_reg)
 }
 
-fn icmp_sig(arg_values: &[TypeArg]) -> Result<FunctionType, SignatureError> {
-    let [arg] = collect_array(arg_values);
-    Ok(FunctionType::new(
-        vec![int_type(arg.clone()); 2],
-        type_row![BOOL_T],
-    ))
+fn btoi_sig(
+    int_type_var: Type,
+    temp_reg: &ExtensionRegistry,
+) -> Result<PolyFuncType, SignatureError> {
+    int_polytype(1, type_row![BOOL_T], vec![int_type_var], temp_reg)
 }
 
-fn ibinop_sig(arg_values: &[TypeArg]) -> Result<FunctionType, SignatureError> {
-    let [arg] = collect_array(arg_values);
-    Ok(FunctionType::new(
-        vec![int_type(arg.clone()); 2],
-        vec![int_type(arg.clone())],
-    ))
+fn icmp_sig(
+    int_type_var: Type,
+    temp_reg: &ExtensionRegistry,
+) -> Result<PolyFuncType, SignatureError> {
+    int_polytype(1, vec![int_type_var; 2], type_row![BOOL_T], temp_reg)
 }
 
-fn iunop_sig(arg_values: &[TypeArg]) -> Result<FunctionType, SignatureError> {
-    let [arg] = collect_array(arg_values);
-    Ok(FunctionType::new(
-        vec![int_type(arg.clone())],
-        vec![int_type(arg.clone())],
-    ))
+fn ibinop_sig(
+    int_type_var: Type,
+    temp_reg: &ExtensionRegistry,
+) -> Result<PolyFuncType, SignatureError> {
+    int_polytype(
+        1,
+        vec![int_type_var.clone(); 2],
+        vec![int_type_var.clone()],
+        temp_reg,
+    )
 }
 
-fn idivmod_checked_sig(arg_values: &[TypeArg]) -> Result<FunctionType, SignatureError> {
-    let [arg0, arg1] = collect_array(arg_values);
-    let intpair: TypeRow = vec![int_type(arg0.clone()), int_type(arg1.clone())].into();
-    Ok(FunctionType::new(
+fn iunop_sig(
```

Comment:
Hmmm, I thought we had renamed these to something more intelligible, like `int_unary_op`. I mean, it makes sense now I've seen it a few times before...